src/DistributedWriteAheadLog/KafkaWALCommon.cpp

```diff
@@ -212,4 +212,42 @@ RecordPtr kafkaMsgToRecord(rd_kafka_message_t * msg, bool copy_topic)
 
     return record;
 }
+
+TopicPartitionStatsPtr
+rdkafkaTopicPartitionStats(rd_kafka_t * rd_kafka, const String & topic, int32_t partition, int timeout_ms)
+{
+    assert(rd_kafka != nullptr);
+    TopicPartitionStatsPtr toppar_stats = std::make_shared<TopicPartitionStats>();
+    rd_kafka_resp_err_t err;
+    do
+    {
```

Comment:
why use do ... while here?

Reply:
break out of `do { something... } while(false)` is a convenient way in C++ used for instead of `goto`